### ISSUE TITLE
Implement new password based Sign In

### ISSUE THREAD

**Issue by Kenzoud** [LookUpGroup27/LookUp#139]
Task description: Introduce a password-based sign-in functionality to allow users to securely access their accounts using an email/username and password combination.

Acceptance criteria:

- Users can sign in to their accounts using a valid email/username and password.
- The password input field is secure and masked to protect user privacy during entry.
- Clear and user-friendly error messages are displayed for invalid credentials or failed sign-in attempts (e.g., "Incorrect email or - 
   password").
- A "Forgot Password?" option is included, redirecting users to a password recovery/reset process.
- The sign-in functionality is secure, using encryption to protect user credentials during transmission and storage.
- The UI for the sign-in process is consistent with the app's overall design and layout.
- The feature is thoroughly tested to ensure successful sign-ins, proper handling of errors, and security compliance.

**Comment by othbcq** [comment]
After reviewing the scope of this issue, I’ve realized that it encompasses multiple distinct functionalities, which makes it too broad to address effectively in a single task. To ensure better clarity, focus, and progress tracking, I will close this issue and split it into the following three separate issues:

1. **Implement Register Feature**: Allow users to create new accounts with email and password, ensuring secure user registration and error handling for invalid inputs.
2. **Implement Login Feature**: Enable users to sign in to their accounts using valid email and password credentials, with clear error messages and secure password masking.
3. **Implement Password Reset Feature**: Provide users with an option to recover/reset their password via email, ensuring a seamless and secure reset process.

This restructuring will help streamline the implementation and testing of each feature while maintaining a clear focus on the requirements.

I will create and link the new issues shortly. Thank you for understanding!

**Comment by othbcq** [comment]
I have created the following new issues:

1. **#169 - Implement Registration Flow**: Focused on allowing users to create new accounts with proper validation and user-friendly UI.
2. **#170 - Implement Login Flow**: Covers the ability for users to securely log in using their email and password, with proper error handling for invalid credentials.
3. **#171 - Implement Password Reset Flow**: Allows users to recover their accounts by sending password reset emails, with feedback on success or failure.

Each issue now targets a specific functionality for easier implementation and testing. Closing this issue in favor of the above three.
